automation/jleechanorg_pr_automation/codex_branch_updater.py

```
@@ -0,0 +1,277 @@
+CREDENTIALS_PATH = Path.home() / ".chatgpt_codex_credentials.json"
+AUTH_STATE_PATH = Path.home() / ".chatgpt_codex_auth_state.json"
+TASK_CARD_SELECTOR = '[data-testid="codex-task-card"]'
+UPDATE_BRANCH_BUTTON_SELECTOR = 'button:has-text("Update Branch")'
+TASK_TITLE_SELECTOR = '[data-testid="codex-task-title"]'
+
+
+def _save_credentials(credentials: Dict[str, str]) -> None:
+    """Persist credentials to disk with restrictive permissions."""
+
+    CREDENTIALS_PATH.write_text(json.dumps(credentials, indent=2))
+    os.chmod(CREDENTIALS_PATH, 0o600)
```

Ruff (0.14.5)

`os.chmod()` should be replaced by `Path.chmod()`

(PTH101)